src/main/java/htsjdk/samtools/cram/build/CramIO.java

```diff
@@ -144,8 +144,13 @@ private static boolean streamEndsWith(final SeekableStream seekableStream, final
     @SuppressWarnings("SimplifiableIfStatement")
     private static boolean checkEOF(final Version version, final SeekableStream seekableStream) throws IOException {
 
-        if (version.compatibleWith(CramVersions.CRAM_v3)) return streamEndsWith(seekableStream, ZERO_F_EOF_MARKER);
-        if (version.compatibleWith(CramVersions.CRAM_v2_1)) return streamEndsWith(seekableStream, ZERO_B_EOF_MARKER);
+        if (version.compatibleWith(CramVersions.CRAM_v3)) {
```

Comment:
Can this be streamlined by having `version` or `CramVersions` return the correct `EOF_MARKER` directly?

Reply:
done